tmt/base.py

```diff
@@ -159,16 +162,16 @@ def run(command):
             remote_name = 'origin'
         remote = run(f"git config --get remote.{remote_name}.url")
         fmf_id['url'] = tmt.utils.public_git_url(remote)
+        if fmf_id['url'] == '' and self.steps() == {'discover'}:
```

Comment:
Again - magic constant `{'discover'}` IMHO empty 'url' could be checked in fmf-id caller (as it matter only there)

Reply:
(same) This code is responsible for the case mentioned by you below https://github.com/teemtee/tmt/pull/932#discussion_r821672134

Reply:
IMHO 'self.steps()' is not necessary and such method doesn't make sense for Test objects. Empty url is problem only for `discover -h fmf --fmf-id` usage and as such should be handled there (e.g. `'url' in fmf_id` check)